### PR TITLE
common.prepare: additional checker to avoid some issues

### PR DESCRIPTION
**Pull request by naumvd95**

### CODE REVIEW

**Review by rkhozinov — APPROVED** [review]

**Review by rkhozinov — CHANGES_REQUESTED** [review]
fix the beginning of the commit to common.prepare:

File: servers/partner-ci/builders/common.prepare.sh [hunk 1]
```diff
@@ -70,28 +70,31 @@ function smart_erase {
...
-        echo "domain destroyed succesfully"
-        virsh undefine --remove-all-storage --snapshots-metadata $vm
-      else
-        ref=$?
-        echo "there are some troubles with virt stack, restart services and recheck"
-	sudo service libvirt-bin restart
+
+  if [ ! -z "$vms" ]; then
+    for vm in $vms; do
+      domstat=$(virsh domstate $vm)
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: `domstat-> domstate`

Comment (rkhozinov) [reply servers/partner-ci/builders/common.prepare.sh]: `domstat=$(virsh domstate $vm)
if [[ $domstat != *"shut"* ]] 
`
to
`
if virsh domstate $vm | grep -e '.*shut.*' -q; then...
`

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/common.prepare.sh [hunk 2]
```diff
@@ -70,28 +70,30 @@ function smart_erase {
...
-      if virsh destroy $vm; then
-        echo "domain destroyed succesfully"
-        virsh undefine --remove-all-storage --snapshots-metadata $vm
-      else
-        ref=$?
-        echo "there are some troubles with virt stack, restart services and recheck"
-	sudo service libvirt-bin restart
+
+  if [ ! -z "$vms" ]; then
+    for vm in $vms; do
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: `if [ ! -z "${vms}" ]; then ... fi `<--remove
 ` for vm in ${vms:?}; `<-- change to

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/common.prepare.sh (hunk 2 above)
Comment (naumvd95) [inline servers/partner-ci/builders/common.prepare.sh]: done

File: servers/partner-ci/builders/common.prepare.sh [hunk 3]
```diff
@@ -70,16 +70,16 @@ function smart_erase {
...
+  for vm in ${vms:?}; do
+    if virsh domstate $vm | grep -e '.*shut.*' -q; then
       if virsh destroy $vm; then
         echo "domain destroyed succesfully"
         virsh undefine --remove-all-storage --snapshots-metadata $vm
       else
         ref=$?
         echo "there are some troubles with virt stack, restart services and recheck"
-	sudo service libvirt-bin restart
+	      sudo service libvirt-bin restart
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: it's seems that the indent is excess

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/common.prepare.sh (hunk 3 above)
Comment (naumvd95) [reply servers/partner-ci/builders/common.prepare.sh]: no, this is github bugs

**Review by rkhozinov — APPROVED** [review]

**Review by rkhozinov — APPROVED** [review]

**Pull request merged**